Mage.Client/src/main/java/mage/client/MageFrame.java

```diff
@@ -746,43 +776,71 @@ public void endDraft(UUID draftId) {
         }
     }
 
-    public void showTournament(UUID tournamentId) {
-        for (Component component : desktopPane.getComponents()) {
-            if (component instanceof TournamentPane
-                    && ((TournamentPane) component).getTournamentId().equals(tournamentId)) {
-                setActive((TournamentPane) component);
-                return;
+    @Override
+    public void showTournament(final UUID tournamentId) {
+        SwingUtilities.invokeLater(() -> {
```

Comment:
I don't know a lot about Java UI. How come you are turning all of these blocks into Lambdas that you pass to `invokeLater`?